File: app/src/main/java/dev/chalinas/priolist/data/ReminderDao.java
package dev.chalinas.priolist.data;

import androidx.room.*
import java.util.List;
import dev.chalinas.priolist.data.Reminder
import kotlinx.coroutines.flow.Flow

@Dao
interface ReminderDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertReminder(reminder: Reminder)

    @Query("SELECT * FROM reminders WHERE taskId = :taskId")
    fun getRemindersForTask(taskId: Int): Flow<List<Reminder>>
}

File: app/src/main/java/dev/chalinas/priolist/data/local/ReminderDao.kt
package dev.chalinas.priolist.data.local

import androidx.room.*
import java.util.List;
import kotlinx.coroutines.flow.Flow

@Dao
interface ReminderDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertReminder(reminder: Reminder)

    @Query("SELECT * FROM reminders WHERE taskId = :taskId")
    fun getRemindersForTask(taskId: Int): Flow<List<Reminder>>
}